src/main/java/edu/ksu/canvas/attendance/submitter/AssignmentSubmitter.java

```diff
@@ -0,0 +1,193 @@
+
+    @Autowired
+    private AttendanceCourseService attendanceCourseService;
+
+    @Autowired
+    private AttendanceService attendanceService;
+
+    @Autowired
+    private CanvasApiWrapperService canvasApiWrapperService;
+
+    @Autowired
+    private AssignmentValidator assignmentValidator;
+
+    @Autowired
+    private CanvasAssignmentAssistant canvasAssignmentAssistant;
+
+
+    /**
+     * This function takes the summary for sections and one by one validates the attendance assignment and
+     * the canvas assignment. Then pushes all the grades and comments to the canvas assignment associated.
+     * @param isSimpleAttendance
+     * @param summaryForSections
+     */
+    public void submitCourseAttendances(boolean isSimpleAttendance, List<AttendanceSummaryModel> summaryForSections, Long courseId,
+                                        OauthToken oauthToken, AttendanceAssignment assignmentConfigurationFromSetup) throws IOException, CanvasOutOfSyncException {
+
+        for (AttendanceSummaryModel sectionSummary : summaryForSections) {
+
+            AttendanceAssignment attendanceAssignment = assignmentService.findBySection(sectionService.getSectionInListById(courseId, sectionSummary.getSectionId()));
+
+            gradePushingValidation(courseId, oauthToken, assignmentConfigurationFromSetup, sectionSummary, attendanceAssignment);
+
+            submitSectionAttendances(isSimpleAttendance, sectionSummary, attendanceAssignment, courseId, oauthToken);
+        }
+    }
+
+    /**
+     * If in validation is determined discrepancy in the data between canvas and teh assignment in the data base,
+     * canvas will be aligned to the data in the db. If in validation is determined that there is not canvas assignment
+     * associated to the attendance assignment then a new canvas assignment will be created and associate to the attendance assignment.
+     */
+    private void gradePushingValidation(Long courseId, OauthToken oauthToken, AttendanceAssignment assignmentConfigurationFromSetup,
+                                        AttendanceSummaryModel model, AttendanceAssignment attendanceAssignment) throws CanvasOutOfSyncException, IOException {
+        try {
+            assignmentValidator.validateConfigurationSetupExistence(model, attendanceAssignment);
+
+            assignmentValidator.validateAttendanceAssignment(courseId, attendanceAssignment, canvasApiWrapperService, oauthToken);
+
+            assignmentValidator.validateCanvasAssignment(assignmentConfigurationFromSetup, courseId, attendanceAssignment, canvasApiWrapperService, oauthToken);
+
+        } catch (CanvasOutOfSyncException exception) {
+
+            if(exception.getMessage().equals("DISCREPANCY BETWEEN CANVAS AND DATABASE")) {
+                canvasAssignmentAssistant.editAssignmentInCanvas(courseId, attendanceAssignment, oauthToken);
+            }
+            else if (exception.getMessage().equals("NO CANVAS ASSIGNMENT LINKED")) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Move the "NO CANVAS ASSIGNMENT LINKED" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1132)